source/slang/slang-emit-base.h

```diff
@@ -0,0 +1,29 @@
+// slang-emit-base.h
+#ifndef SLANG_EMIT_BASE_H
+#define SLANG_EMIT_BASE_H
+
+#include "../core/slang-basic.h"
+
+#include "slang-ir.h"
+#include "slang-ir-insts.h"
+#include "slang-ir-restructure.h"
+
+namespace Slang
+{
+
+class SourceEmitterBase : public RefObject
```

Comment:
Should this be `EmitterBase` since for SPIR-V we aren't emitting source? I guess I'm jumping ahead, but I'm just *assuming* that's how it will be used...